core/src/main/java/apoc/merge/Merge.java

```diff
@@ -83,16 +82,20 @@ public Stream<NodeResultWithStats> nodeWithStats(@Name("labels") List<String> la
     }
 
     private Result getNodeResult(List<String> labelNames, Map<String, Object> identProps, Map<String, Object> props, Map<String, Object> onMatchProps) {
-        if (identProps ==null || identProps.isEmpty()) {
+        if (identProps == null || identProps.isEmpty()) {
             throw new IllegalArgumentException("you need to supply at least one identifying property for a merge");
         }
 
-        String labels = labelString(labelNames);
+        if (labelNames != null && (labelNames.contains(null) || labelNames.contains(""))) {
+            throw new IllegalArgumentException("The list of label names may not contain any null or empty String values. If you wish to merge a node without a label, pass an empty list instead.");
+        }
```

Comment:
I don't know if we want to use the new types here like NODE, LIST and STRING, maybe check with @gem-neo4j as she has worked hard to unify the use of them.

Same applies to the relationship error message

Reply:
I see the update and will add my comment here to continue the thread:

"The list of label names may not contain any `NULL` or empty `STRING` values. If you wish to merge a `NODE` without a label, pass an empty `LIST` instead."

The first list has been missed, and the second time list is mentioned, it mentions `LIST` by itself, which is not a valid Cypher Type name.

I am also unsure about our take on error messages containing personal pronouns?

Reply:
There are some new guidelines for error messages but we will discuss those in a meeting in 35 minutes so they are not really official/well structured yet

Reply:
This is WIP PR of error guidelines if you are curious 
https://github.com/neo4j/docs-status-codes/pull/73

Reply:
How about this?
"The list of label names may not contain any NULL or empty STRING values. If you wish to merge a NODE without a label, pass an empty list instead."

Reply:
I think that is OK, do you agree @gem-neo4j ?

Reply:
Yeah, I think that is fine :)